variables.tf

```diff
@@ -141,6 +141,23 @@ variable "system_assigned_identity_enabled" {
   default     = false
 }
 
+variable "immutability_policy" {
+  description = "An immutability policy to configure for this Storage account."
+
+  type = object({
+    state                         = optional(string, "Unlocked")
+    allow_protected_append_writes = optional(bool, false)
+    period_since_creation_in_days = optional(number, 30)
```

Comment:
The other default values are inspired by the Azure Portal default values, however I don't have a good reasoning here.